src/lib.rs

```diff
@@ -227,17 +237,23 @@ impl KafkaJsonToDelta {
     async fn init(&mut self, state: &ProcessingState) -> Result<(), KafkaJsonToDeltaError> {
         info!("Initializing last Delta transaction version before starting run loop.");
 
-        let last_txn_version = state
-            .delta_writer
-            .last_transaction_version(self.app_id.as_str())
-            .await?;
+        let mut offsets = HashMap::new();
 
-        if let Some(txn_version) = last_txn_version {
-            let previous_wal_entry = state.wal.get_entry_by_transaction_id(txn_version).await?;
-            self.init_offsets(&previous_wal_entry.partition_offsets)
+        for p in state.value_buffers.partitions.iter() {
+            let partition = *p;
+
+            let last_txn_version = state
+                .delta_writer
+                .last_transaction_version(&self.app_id_for_partition(partition))
```

Comment:
not related to your change, but I think `last_transaction_version` doesn't need to be defined as an async method.